src/Sparrow/Json/BlittableJsonReaderArray.cs

```diff
@@ -104,7 +107,11 @@ public BlittableJsonReaderArray Clone(JsonOperationContext context, BlittableJso
         public void Dispose()
         {
             AssertContextNotDisposed();
-            _parent?.Dispose();
+
+            // this is required only in cases that we get a BlittableJsonReaderArray, which is an only child of an BlittableJsonReaderObject and we lose track of it's parent, 
+            // like in BlittableJsonDocumentBuilder.CreateArrayReader.
+            if (_disposeParent)
+                _parent?.Dispose();
```

Comment:
This introduces almost totally opposite behavior to the current one. Instead of always disposing the parent, we do it _only_ in one particular case. Maybe that's ok, I'm just a bit concerned about implication and potential memory issues.

Can you explain, please?

Reply:
The reason this feature existed is to handle the `CreateArrayReader`, where we don't *have* access to the parent blittable.
Otherwise, there is no point in trying to dispose a blittable that is an inner value.